types/global.d.ts

```diff
@@ -0,0 +1,41 @@
+import { Request, Response, NextFunction } from "express";
+import { Boom } from "express-boom";
+
+export type userData = {
+  message: string;
+  user: {
+    id: string;
+    profileURL: string;
+    discordJoinedAt: string;
+    roles: {
+      archived: boolean;
+      in_discord: boolean;
+      member: boolean;
+    };
+    profileStatus: string;
+    created_at: number;
+    yoe: number;
+    github_created_at: number;
+    company: string;
+    twitter_id: string;
+    first_name: string;
+    incompleteUserDetails: boolean;
+    discordId: string;
+    last_name: string;
+    linkedin_id: string;
+    picture: {
+      url: string;
+      publicId: string;
+    };
+    instagram_id: string;
+    github_display_name: string;
+    github_id: string;
+    designation: string;
+    status: string;
+    username: string;
+    updated_at: string;
+  };
+};
+
+export type CustomResponse = Response & { boom: Boom };
```

Comment:
what is & here, bitwise and?

Reply:
so this is used in typescript to create one type which has multiple types, so the original request doesn't have boom object in it, we are using a package for that because of that I am using a intersection type here read this for more info https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#:~:text=An%20intersection%20type%20combines%20multiple,Person%20and%20Serializable%20and%20Loggable%20.

Reply:
got it